### DISCUSSION TITLE
Darstellungsfilter ohne Platzhalter

### DISCUSSION THREAD

**Discussion by k-placer** [e-netze/webgis-community#35] [General]
Wenn ich einen Darstellungsfilter definiere, der keinen Parameter verwendet:
z.B. SQL Filter: 
`AUSSENSTELLE_NAME =  'BH'`
 
Erscheint eine Fehlermeldung im Viewer. 
<img width="505" alt="darstellungsfilter_error" src="https://user-images.githubusercontent.com/97947925/216934025-d8a361f2-dcd1-4a96-931c-b0b0d95a82f6.PNG">

So funktioniert der Darstellungsfilter problemlos:
SQL Filter: 
`AUSSENSTELLE_NAME = 'BH' or AUSSENSTELLE_NAME ='[Aussenstelle]'`

Warum muss hier zwingend ein Platzhalter in den SQL-Filter eingetragen werden?

**Comment by gis-eni** [comment]
Das hier eine Fehlermeldung kommt, kann man natürlich als **Bug** deklarieren. Sollte nicht so sein und der Fehler wird behoben.
Anscheinend hat noch niemand (auch nicht bei den Tests) probiert, ob das ohne Parameter funktioniert. Und das seit WebGIS 2.0? ;)

Die Frage ist für mich aber: Warum macht man das?
Filter ohne *Parameter* (also ohne direkten Zusammenhang) ziehen natürlich einen Rattenschwanz an Überlegungen und Änderungen nach sich:
* Soll so ein Filter zum Setzen auch in der Ergbnistabelle, Ergebnis-Details angezeigt werden?
* Was, wenn mehrere "statische" Filter auf ein Thema definiert werden. Wenn der Anwender jetzt schon einen Filter auf ein Thema gesetzt hat, und danach noch einen weiteren, dann sieht er gar nix mehr, wenn sich beide Filter gegenseitig ausschließen. Währe eine Auswahlliste mit Werten da nicht besser?
* Handelt es sich um gesperrte Filter, die der Anwender gar nicht setzten kann? Wäre es da nicht sowieso besser, schon in der MXD zu filtern.
* Sollen sollte Filter auch per Url Parameter übergeben werden?
* ...

Vielleicht kannst du hier noch einmal kurz erklären, was der Anwendungsfall für solche Filter ist?

**Comment by gis-eni** [comment]
Das Problem ist mit Build 5.23.601 (bereits verfügbar) behoben.
Leider hast du nicht mehr beschreiben, wozu du diese Art der Filter benötigst. Ich hoffe man es funktioniert trotzdem...
